model/execution/KVDeliveryExecution.php

```diff
@@ -66,6 +66,14 @@ public function getIdentifier()
         return $this->id;
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function getOriginalIdentifier(): string
+    {
+        return str_replace(KeyValueService::DELIVERY_EXECUTION_PREFIX, '', $this->id);
```

Comment:
This is still unsafe.

```suggestion
        return substr($this->id, strlen(KeyValueService::DELIVERY_EXECUTION_PREFIX));
```

Same applies to the RDS storage repository.